stripecardscan/src/main/java/com/stripe/android/stripecardscan/cardscan/CardScanActivity.kt

```diff
@@ -0,0 +1,683 @@
+
+    private fun ensureValidParams() = when {
+        params.stripePublishableKey.isEmpty() -> {
+            scanFailure(InvalidStripePublishableKeyException("Missing publishable key"))
+            false
+        }
+//        params.cardImageVerificationIntentId.isEmpty() -> {
+//            scanFailure(InvalidCivException("Missing card image verification ID"))
+//            false
+//        }
+//        params.cardImageVerificationIntentSecret.isEmpty() -> {
+//            scanFailure(InvalidCivException("Missing card image verification client secret"))
+//            false
+//        }
+        else -> true
+    }
+
+    private fun closeScanner() {
+//        uploadScanStats(
+//            stripePublishableKey = params.stripePublishableKey,
+//            civId = params.cardImageVerificationIntentId,
+//            civSecret = params.cardImageVerificationIntentSecret,
+//            instanceId = Stats.instanceId,
+//            scanId = Stats.scanId,
+//            device = Device.fromContext(this),
+//            appDetails = AppDetails.fromContext(this),
+//            scanStatistics = ScanStatistics.fromStats()
+//        )
+        setFlashlightState(false)
+        finish()
+    }
+
+    private fun displayState(newState: ScanState, previousState: ScanState?) {
+        when (newState) {
+            is ScanState.NotFound -> {
+                viewBinding.viewFinderBackground
+                    .setBackgroundColor(getColorByRes(R.color.stripeNotFoundBackground))
+                viewBinding.viewFinderWindow
+                    .setBackgroundResource(R.drawable.stripe_card_background_not_found)
+                viewBinding.viewFinderBorder.startAnimation(R.drawable.stripe_card_border_not_found)
+                viewBinding.instructions.setText(R.string.stripe_card_scan_instructions)
+            }
+            is ScanState.FoundShort -> {
+                viewBinding.viewFinderBackground
+                    .setBackgroundColor(getColorByRes(R.color.stripeFoundBackground))
+                viewBinding.viewFinderWindow
+                    .setBackgroundResource(R.drawable.stripe_card_background_found)
+                viewBinding.viewFinderBorder.startAnimation(R.drawable.stripe_card_border_found)
+                viewBinding.instructions.setText(R.string.stripe_card_scan_instructions)
+                viewBinding.instructions.show()
+            }
+            is ScanState.FoundLong -> {
+                viewBinding.viewFinderBackground
+                    .setBackgroundColor(getColorByRes(R.color.stripeFoundBackground))
+                viewBinding.viewFinderWindow
+                    .setBackgroundResource(R.drawable.stripe_card_background_found)
+                viewBinding.viewFinderBorder
+                    .startAnimation(R.drawable.stripe_card_border_found_long)
+                viewBinding.instructions.setText(R.string.stripe_card_scan_instructions)
+                viewBinding.instructions.show()
+            }
```

Comment:
We should determine which of these is appropriate for `Found`. The difference is the length of time the animation takes to circle the view finder. We can determine what this time should be based on the state machine.